Draft: Update the README file (#1)
sergepilipchuk/maui-theme-selector-app: README.md

* [DXBorder](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXBorder): [BackgroundColor](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.visualelement.backgroundcolor?view=net-maui-8.0), [CornerRadius](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXViewBase.CornerRadius)
* [DXCollectionView](https://docs.devexpress.com/MAUI/DevExpress.Maui.CollectionView.DXCollectionView): [SelectedItemTemplate](https://docs.devexpress.com/MAUI/https://docs.devexpress.com/MAUI/DevExpress.Maui.CollectionView.DXCollectionView.SelectedItemTemplate), [ItemsSource](https://docs.devexpress.com/MAUI/DevExpress.Maui.CollectionView.DXCollectionView.ItemsSource), [ItemTemplate](https://docs.devexpress.com/MAUI/DevExpress.Maui.CollectionView.DXCollectionView.ItemTemplate)
* [DXButton](https://docs.devexpress.devx/MAUI/DevExpress.Maui.Core.DXButton): [Content](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXBorder.Content), [ButtonType](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXButton.ButtonType)

## Implementation Details

> 1. Use the [ChoiceChipGroup](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChoiceChipGroup) control to display and group round color markers that allow you to select a new color. To specify a round color marker, use the [DXBorder](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXBorder) control.

Reviewer: ```suggestion
1. Use the [ChoiceChipGroup](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChoiceChipGroup) control to display and group round color markers that allow you to select a new color. To add a round color marker, use the [DXBorder](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXBorder) control.
```

    ```xaml
    <dx:ChoiceChipGroup 
        SelectedItem="{Binding SelectedColor, Mode=TwoWay}" 
        ItemsSource="{Binding Items}" 
        IsMultiline="True"
        ... >
        <dx:ChoiceChipGroup.ChipContentTemplate>
            <DataTemplate>
                <dx:DXBorder
                    CornerRadius="26" 
                    WidthRequest="52" 
                    HeightRequest="52"
                    BackgroundColor="{Binding Color}">
                    <Label Text="{Binding Name}" ... />
                </dx:DXBorder>
            </DataTemplate>
        </dx:ChoiceChipGroup.ChipContentTemplate>
    </dx:ChoiceChipGroup>
    ```

> 2. Use the `ThemeManager.GetSeedColor()` method to get a [seed color](https://docs.devexpress.com/MAUI/404636/common-concepts/themes#create-a-theme-based-on-a-custom-seed-color).

Reviewer: I'm not sure why this description focuses on `GetSeedColor()`. There's a lot of stuff going on in code that we can talk about. The main thing, I guess, is to link this to the overall flow. 

Since in the last step the code defines a `ChoiceChipGroup` bound to the `Items` collection, we should probably mention here that this is that data source. 

```suggestion
2. As you see in the previous step, the `ChoiceChipGroup` control obtains data from the view model's `Items` collection. Populate this collection with pre-defined colors. Use the `ThemeManager.GetSeedColor()` method to obtain [seed colors](https://docs.devexpress.com/MAUI/404636/common-concepts/themes#create-a-theme-based-on-a-custom-seed-color).
```

    ```csharp
    public ThemesViewModel() {
        Items = new ObservableCollection<ColorModel>() {
    #if ANDROID
            new ColorModel(Colors.Black, "System Color", true),
    #endif

>             new ColorModel(ThemeManager.GetSeedColor(ThemeSeedColor.Purple), ThemeSeedColor.Purple.ToString()),

Reviewer: I would probably add at least one more line. Right now, it almost looks like there's either ANDROID or other OS. And they options get one color each. 

```suggestion
            new ColorModel(ThemeManager.GetSeedColor(ThemeSeedColor.Purple), ThemeSeedColor.Purple.ToString()),
            new ColorModel(ThemeManager.GetSeedColor(ThemeSeedColor.Violet), ThemeSeedColor.Violet.ToString()),
```

    public ThemesViewModel() {
        Items = new ObservableCollection<ColorModel>() {
    #if ANDROID
            new ColorModel(Colors.Black, "System Color", true),
    #endif
            new ColorModel(ThemeManager.GetSeedColor(ThemeSeedColor.Purple), ThemeSeedColor.Purple.ToString()),
            new ColorModel(ThemeManager.GetSeedColor(ThemeSeedColor.Violet), ThemeSeedColor.Violet.ToString()),
            // ...
        };
        SelectedColor = Items[0];
        SelectedThemeType = ThemeTypes[0];
    }
    ``` 

3. Use the `new Theme()` constructor to create a new theme based on the selected color.

    ```csharp
    [ObservableProperty]
    ColorModel selectedColor;

    partial void OnSelectedColorChanged(ColorModel value) {
        if (value == null)
            return;

        PreviewColorName = value.DisplayName;
        if (value.IsSystemColor) {
            // Use the color from the Android system
            ThemeManager.UseAndroidSystemColor = true;
            return;
        }
        
        ThemeManager.UseAndroidSystemColor = false;
        // Creates a new theme based on the selected color
        ThemeManager.Theme = new Theme(value.Color);
    }    

> 4. Use the [`DXCollectionView`](https://docs.devexpress.com/MAUI/DevExpress.Maui.CollectionView.DXCollectionView) control to display radio buttons that allow you to select a theme type (*dark/light/system*). Specify the data source and item templates (use different templates for selected and deselected item).

Reviewer 2: ```suggestion
4. Use the [`DXCollectionView`](https://docs.devexpress.com/MAUI/DevExpress.Maui.CollectionView.DXCollectionView) control to display radio buttons that allow you to select a theme type (*dark/light/system*). Specify the data source and item templates (use different templates for selected and deselected items).
```
